### PR TITLE
Use cloneDeep() to make diagramOptions immutable

### LINKED ISSUE #753: FHI Angular Highcharts: gjør diagramOptions immutable
### Description

I dag legges det til properties på `FhiDiagramOptions` til internt bruk i `FhiAngularHighchartsComponent`. Dette er forvirrende for konsumenter av komponenten siden endringene også reflekteres utenfor komponenten, det er samme objekt. En bedre løsning vil være å klone `FhiDiagramOptions`-objektet, kun endre kopien internt, og aldri eksponere denne kopien på utsiden av komponenten.

### Acceptance Criteria

- Det skal ikke lenger skje at objektet `@Input() diagramOptions` endres på innsiden av `FhiAngularHighchartsComponent`.
- Utover dette skal denne PR-en ikke ha noen observerbare "side effects".

### PR DESCRIPTION
**Pull request by proand**
This closes #753

### CODE REVIEW

File: projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<article *ngIf="diagramOptions !== undefined">
-  <h1 class="h3 mb-3">{{ diagramOptions.title }}</h1>
+<article *ngIf="diagramOptionsInternal !== undefined">
+  <h1 class="h3 mb-3">{{ diagramOptionsInternal.title }}</h1>
 
-  @if (diagramOptions.description) {
-    <p class="mb-4">{{ diagramOptions.description }}</p>
+  @if (diagramOptionsInternal.description) {
```
Comment (VillereV) [inline projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html:4]: @proand A combination of *ngIf and @if in one file is a bit untidy.

File: projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
-<article *ngIf="diagramOptions !== undefined">
-  <h1 class="h3 mb-3">{{ diagramOptions.title }}</h1>
+<article *ngIf="diagramOptionsInternal !== undefined">
```
Comment (VillereV) [inline projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html:1]: @proand The user might send diagramOption as `null`. Maybe change `diagramOptionsInternal !== undefined` 
`@if(diagramOptionsInternal)` or
`@if(diagramOptionsInternal != null)` or
`@if(diagramOptionsInternal !== null || diagramOptionsInternal !== undefined)`

File: projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html [hunk 3]
```diff
@@ -78,16 +78,16 @@ <h1 class="h3 mb-3">{{ diagramOptions.title }}</h1>
...
     [size]="'fullscreen'"
   >
     <ng-container fhi-modal.button>
       <i class="icon-arrows-fullscreen"></i>
       <span class="visually-hidden">Fullskjerm </span>
     </ng-container>
     <ng-container fhi-modal.body>
-      @if (diagramOptions.description) {
-        <p>{{ diagramOptions.description }}</p>
+      @if (diagramOptionsInternal.description) {
```
Comment (VillereV) [inline projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html:89]: @proand  `description` property is optional. Use `@if (diagramOptionsInternal?.description)`

File: projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html (hunk 1 above)
Comment (proand) [reply projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html:4]: Created a new issue to handle this: #768

File: projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html (hunk 2 above)
Comment (proand) [reply projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.html:1]: I have a little list (not exhaustive) of values, or missing values, in the diagramOptions that can give errors in current implementation. I'm pondering making a test of the input, and then jsut test for `!diagramOptionsValid` or something like that... but probably best to create an issue for that, and go for your solution now.

**Review by bittakv — APPROVED** [review]
LGTM!
The diagramOptions provided as imput to the component now seem to be untouched by FhiAngularHighchartsComponent :-)

**Pull request merged**